### PR TITLE
Add SimpleTests; add GTest extension; fix and test `abs()`.

### PR DESCRIPTION
**Pull request by JoshDreamland**
Also disables ASAN return code overrides, as this will make all tests fail until all memory leaks are fixed.

### CODE REVIEW

**Comment by fundies** [comment]
looks like it's only generating or finding one coverage report

File: CommandLine/testing/SanityCheck.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    if (others.size() == 1) {
+      FAIL() << "Junk file \"" << others.begin()->first << '"';
+    } else {
+      FAIL() << "Found " << others.size() << " junk files, including "
+             << '"' << others.begin()->first << '"';
+    }
+  }
+}
+
+void bitch_about_junk_files(const NameMap &junk) {
```
Comment (fundies) [inline CommandLine/testing/SanityCheck.cpp]: maybe bitch_about -> print

File: CommandLine/testing/SimpleTests/math_test.sog/create.edl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+float f = (float) -.5;
```
Comment (fundies) [inline CommandLine/testing/SimpleTests/math_test.sog/create.edl]: does enigm support 3.f ?

File: CommandLine/testing/SimpleTests/math_test.sog/create.edl [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// long double ld = -.5;
+int i = -30;
+unsigned u = 30;
+long l = -60;
+unsigned lu = 60;
+variant v = -120;
+var a = -150;
+
+cons_show_message("Test start!");
+gtest_assert_eq(abs(f),  -f);
```
Comment (fundies) [inline CommandLine/testing/SimpleTests/math_test.sog/create.edl]: I'm no seeing anything about the tests between the start / ens mesages in the log

File: ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/include.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#ifndef ENIGMA_EXTENSION_GTEST_h
```
Comment (fundies) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/include.h]: I think we're going with capital H(s)

File: CommandLine/testing/SanityCheck.cpp [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+#include <gtest/gtest.h>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "TestHarness.hpp"
+
+namespace {
+
```
Comment (fundies) [inline CommandLine/testing/SanityCheck.cpp]: y so many empty lines

File: CommandLine/testing/SanityCheck.cpp [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+using std::map;
+using std::string;
+using std::vector;
+typedef map<string, string> NameMap;
```
Comment (fundies) [inline CommandLine/testing/SanityCheck.cpp]: perhaps using here since we're using c++11

File: CommandLine/testing/SanityCheck.cpp [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      games->insert(NameMap::value_type(fullname, filename));
+    } else if (sources && (ext == "cpp" || ext == "cc")) {
+      sources->insert(NameMap::value_type(fullname, filename));
+    } else if (others) {
+      others->insert(NameMap::value_type(fullname, filename));
+    }
+  }
+}
+
+using std::cout;
```
Comment (fundies) [inline CommandLine/testing/SanityCheck.cpp]: group at top with other usings

File: ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/GTest.cpp [hunk 8]
```diff
@@ -0,0 +1,36 @@
+#include "include.h"
+#include <gtest/gtest.h>
+using std::string;
```
Comment (fundies) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/GTest.cpp]: new line between includes and code

File: CommandLine/testing/SimpleTests/math_test.sog/create.edl (hunk 3 above)
Comment (fundies) [inline CommandLine/testing/SimpleTests/math_test.sog/create.edl]: I think real numbers should be in the results column here so we don't end up in a 0 == 0 situation without realizing it

File: ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/About.ey [hunk 9]
```diff
@@ -0,0 +1,12 @@
+%e-yaml
+---
+
+Name: Google Test
+Identifier: GTest
+Author: Josh
+Description: Exposes Google Test checks through an ENIGMA API. (Including Google Test directly would break everything.)
+Default: false
+Icon: beaker.png
```
Comment (fundies) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/About.ey]: you draw this yourself?

File: CommandLine/testing/SanityCheck.cpp [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+typedef map<string, string> NameMap;
+
+const char *const kSimpleTestDirectory = "CommandLine/testing/SimpleTests";
+const char *const kDrivenTestDirectory = "CommandLine/testing/Tests";
+
+void read_files(string directory,
+                NameMap *games, NameMap *sources, NameMap *others) {
+  boost::filesystem::path targetDir(directory);
+  boost::filesystem::directory_iterator iter(targetDir), eod;
+  BOOST_FOREACH(boost::filesystem::path const& i, std::make_pair(iter, eod)) {
```
Comment (fundies) [inline CommandLine/testing/SanityCheck.cpp]: boost has a .stem and .extension function on paths

**Comment by fundies** [comment]
google test has a flag to output results as xml. perphaps we could enable that and write a simple php page to show cleaner results

File: CommandLine/testing/SimpleTests/math_test.sog/create.edl (hunk 2 above)
Comment (RobertBColton) [reply CommandLine/testing/SimpleTests/math_test.sog/create.edl]: Just write 3f or 3.0f if you want to be pedantic.

File: ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/About.ey (hunk 9 above)
Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/GTest/About.ey]: He always does.

**Comment by RobertBColton** [comment]
> Do not try and bend the fundies, that's impossible. Instead, only try to realize the truth...there is no fundies. Then you'll see that it is not the fundies that bends, it is only yourself.

**Comment by JoshDreamland** [comment]
This change increases our code coverage by 0.15%. It also makes it actually work again.

**Comment by JoshDreamland** [comment]
It's in #1181.

**Pull request merged**